tests/x11/yast2_snapper.pm

```diff
@@ -51,8 +51,13 @@ sub clean_and_quit() {
     if (!$ret) {
         $self->{mute_post_fail} = 1;
         record_soft_failure 'bsc#1032831';
-        $self->failure_analysis();
-        assert_screen 'root-gnome-terminal', 90;
+        # Wait another 90 seconds for the commands to finish
+        my $ret = check_screen('root-gnome-terminal', 90);
+        if (!$ret) {
+            $self->{mute_post_fail} = 1;
```

Comment:
You can get rid of $ret and using unless to avoid negation: 
unless (check_screen).  
We also could refactor code above to do same. It's also tricky if you define two variables with same name, as in case of visibility change we may get a conflict.